### PR TITLE
[eclipse/xtext#1679]ported more code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679]ported more code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	public void fireproof(CharSequence input) throws Exception {
+		fireproof(input.toString());
+	}
+
+	public void fireproof(String input) throws Exception {
+		try {
+			EObject file = parseHelper.parse(input);
+			IResolvedTypes resolvedTypes = typeResolver.resolveTypes(file);
+			Assert.assertNotNull(resolvedTypes);
+			if (file != null) {
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:66]: Reverse logic and early return?

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+				for (EObject content : IteratorExtensions.toIterable(file.eAllContents())) {
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:67]: Ideally we use something other than IteratorExtensions

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+					if (content instanceof XAbstractFeatureCall) {
+						assertExpressionTypeIsResolved(((XExpression) content), resolvedTypes);
+						if (((XAbstractFeatureCall) content).getImplicitReceiver() != null) {
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:70]: Might want to make a casted var for the XAbstractFeatureCall

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+							assertExpressionTypeIsResolved(((XAbstractFeatureCall) content).getImplicitReceiver(), resolvedTypes);
+						}
+						if (((XAbstractFeatureCall) content).getImplicitFirstArgument() != null) {
+							assertExpressionTypeIsResolved(((XAbstractFeatureCall) content).getImplicitFirstArgument(), resolvedTypes);
+						}
+					} else if (content instanceof XClosure) {
+						assertExpressionTypeIsResolved(((XExpression) content), resolvedTypes);
+						((XClosure) content).getImplicitFormalParameters()
+								.forEach((Consumer<JvmFormalParameter>) (JvmFormalParameter it) -> {
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:79]: Unneeded cast?

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+ */
+public abstract class AbstractXbaseContentAssistBugTest extends AbstractXbaseUITestCase implements IJavaProjectProvider {
+	private IProject demandCreateProject;
+
+	private static IProject staticProject;
+
+	@BeforeClass
+	public static void createTestProject() throws Exception {
+		TargetPlatformUtil.setTargetPlatform(AbstractXbaseContentAssistBugTest.class);
+		AbstractXbaseContentAssistBugTest.staticProject = AbstractXbaseUITestCase
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java:40]: No need for qualified name

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+				.createPluginProject(AbstractXbaseContentAssistBugTest.class.getName());
+	}
+
+	@AfterClass
+	public static void deleteTestProject() throws Exception {
+		JavaProjectSetupUtil.deleteProject(AbstractXbaseContentAssistBugTest.staticProject);
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java:46]: No need for qualified name

File: org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		return false;
+	}
+
+	@Override
+	public IJavaProject getJavaProject(ResourceSet resourceSet) {
+		String projectName = getProjectName();
+		IJavaProject javaProject = JavaProjectSetupUtil.findJavaProject(projectName);
+		if (javaProject == null || !javaProject.exists()) {
+			try {
+				demandCreateProject = AbstractXbaseUITestCase.createPluginProject(projectName);
```
Comment (tivervac) [inline org.eclipse.xtext.xbase.ui.tests/src/org/eclipse/xtext/xbase/ui/tests/editor/AbstractXbaseContentAssistBugTest.java:68]: No need for qualified name

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/XbaseIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.xtext.xbase.testing.OnTheFlyJavaCompiler2;
+import org.eclipse.xtext.xbase.testing.evaluation.AbstractXbaseEvaluationTest;
+import org.junit.runner.RunWith;
+
+import com.google.inject.Inject;
+
+/**
+ * Xbase integration test.
+ * 
+ * runs all Xbase tests from {@link AbstractXbaseEvaluationTest} in the context
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.tests/src/org/eclipse/xtext/example/domainmodel/tests/XbaseIntegrationTest.java:29]: s/runs/Runs/

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyBuilder.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+	private AssociationHierarchyNodeLocationProvider associationHierarchyNodeLocationProvider;
+
+	@Override
+	protected IHierarchyNodeLocationProvider getHierarchyNodeLocationProvider() {
+		return associationHierarchyNodeLocationProvider;
+	}
+
+	@Override
+	protected IEObjectDescription findDeclaration(URI objectURI) {
+		final IEObjectDescription description = getDescription(objectURI);
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyBuilder.java:45]: No need for final

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyBuilder.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+			}
+		}
+		return readOnly(objectURI, (EObject object) -> {
+			return getDescription(EcoreUtil2.getContainerOfType(object, Entity.class));
+		});
+	}
+
+	@Override
+	protected boolean filterReference(IReferenceDescription reference) {
+		if (reference == null || !this.isJvmType(reference.getEReference().getEType())) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyBuilder.java:70]: No need for `this`

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java [hunk 11]
```diff
@@ -16,14 +16,13 @@
...
 public class AssociationHierarchyViewPart extends DefaultCallHierarchyViewPart {
-  @Override
-  protected Pair<String, ColumnLayoutData>[] getLocationColumnDescriptions() {
-    ColumnWeightData _columnWeightData = new ColumnWeightData(60);
-    Pair<String, ColumnLayoutData> _mappedTo = Pair.<String, ColumnLayoutData>of("Line", _columnWeightData);
-    ColumnWeightData _columnWeightData_1 = new ColumnWeightData(300);
-    Pair<String, ColumnLayoutData> _mappedTo_1 = Pair.<String, ColumnLayoutData>of("Property", _columnWeightData_1);
-    return new Pair[] { _mappedTo, _mappedTo_1 };
-  }
+	@SuppressWarnings("unchecked")
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java:20]: Don't really see why this is needed

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java [hunk 12]
```diff
@@ -16,14 +16,13 @@
...
+	@Override
+	protected Pair<String, ColumnLayoutData>[] getLocationColumnDescriptions() {
+		return new Pair[] { 
+			Pair.<String, ColumnLayoutData> of("Line", new ColumnWeightData(60)),
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java:24]: Don't need the typing information right?

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java (hunk 11 above)
Comment (szarnekow) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/editor/hierarchy/AssociationHierarchyViewPart.java:20]: Generic array creation.

**Comment by cdietrich** [comment]
@tivervac which encoding do you refer to

**Comment by tivervac** [comment]
@cdietrich Huh, I was reviewing with gitpod, seems like it made changes to the .project files and I thought those were your changes. Disregard my comment about encodings.

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+		Assert.assertTrue(indicator.isCanceled());
+		document.internalModify((XtextResource it) -> {
+			Assert.assertFalse(document.getCancelIndicator().isCanceled());
+			return null;
+		});
+	}
+
+	@Test
+	public void testPriorityReadOnlyCancelsReaders() throws Exception {
+		Thread.interrupted();
```
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java:91]: i did not analyze the sematics of the test and i dont want to

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:67]: dont know an alternative

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:66]: wont spent too much optimization on dreprecated class

**Comment by cdietrich** [comment]
@tivervac anything left?

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (tivervac) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java:86]: I still think this should be removed

[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java:86]: as i said. i have no idea what it does. and dont want to change it
@szarnekow what do you think

File: org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java (hunk 1 above)
Comment (tivervac) [reply org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java:65]: It's more about the readability to me

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java:86]: see https://github.com/eclipse/xtext-eclipse/commit/cfcf1ec20cd899da48c8044ac5957d24cae21674
will take over the comment

[the remark this answers is not in the document]
Comment (tivervac) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/model/DocumentLockerTest.java]: I thought this was just a getter, seems like it isn't so just leave it in.
From `Thread`'s Javadoc:

`The <i>interrupted status</i> of the thread is cleared by this method.`

**Comment by cdietrich** [comment]
@tivervac @szarnekow  ok now?
want to have this in M2 tomorrow

**Pull request merged**